src/core/ext/filters/client_channel/retry_throttle.cc

```diff
@@ -30,184 +30,156 @@
 
 #include "src/core/lib/avl/avl.h"
 
+namespace grpc_core {
+namespace internal {
+
 //
-// server_retry_throttle_data
+// ServerRetryThrottleData
 //
 
-struct grpc_server_retry_throttle_data {
-  gpr_refcount refs;
-  int max_milli_tokens;
-  int milli_token_ratio;
-  gpr_atm milli_tokens;
-  // A pointer to the replacement for this grpc_server_retry_throttle_data
-  // entry.  If non-nullptr, then this entry is stale and must not be used.
-  // We hold a reference to the replacement.
-  gpr_atm replacement;
-};
-
-static void get_replacement_throttle_data_if_needed(
-    grpc_server_retry_throttle_data** throttle_data) {
+ServerRetryThrottleData::ServerRetryThrottleData(
+    int max_milli_tokens, int milli_token_ratio,
+    ServerRetryThrottleData* old_throttle_data)
+    : max_milli_tokens_(max_milli_tokens),
+      milli_token_ratio_(milli_token_ratio) {
+  int initial_milli_tokens = max_milli_tokens;
```

Comment:
we should use `gpr_atm` (aka `intptr_t`, but that's an implementation detail) for anything that ends up used in atomic operations. There are no guarantees that a regular `int` will be the right size on some platforms, thus `intptr_t`.

It seems this was a pre-existing issue, but an issue nonetheless.